### PR TITLE
Crolke 918 vhost hostname

### PR DESCRIPTION
**Pull request by ChugR**

### CODE REVIEW

**Comment by ChugR** [comment]
This PR adds vhost attribute 'hostname' as the name of the entity. It deprecates vhost attribute 'id'.
Policy code warns if 'id' is used and fails if neither 'id' nor 'hostname' is specified.

File: console/config/schema.json [hunk 1]
```diff
@@ -1273,6 +1273,10 @@
           "type": "map", 
           "description": "A map where each key is a user group name and the value is a map of the corresponding settings for that group."
         }, 
+        "hostname": {
```
Comment (ganeshmurthy) [inline console/config/schema.json]: Could this use the deprecationName attribute from my earlier PR - https://github.com/apache/qpid-dispatch/pull/261/files#diff-981a10a2c9cab3708f003b402e7f8be4R1060

File: console/config/schema.json [hunk 2]
```diff
@@ -1273,6 +1273,10 @@
...
+          "type": "string", 
+          "description": "The vhost name."
+        },
```
Comment (bhardesty) [inline console/config/schema.json]: In PR 255, I updated the policy attribute descriptions (including "id") to be more descriptive. For the hostname description, you could use something like this:

The hostname of the vhost. This vhost policy will be applied to any client connection that is directed to the hostname that you specify.

File: console/config/schema.json (hunk 1 above)
Comment (ChugR) [reply console/config/schema.json]: Can't be used until you check it in...

Comment (ganeshmurthy) [reply console/config/schema.json]: yes, agreed

File: console/config/schema.json (hunk 2 above)
Comment (ChugR) [reply console/config/schema.json]: will do.

**Comment by ChugR** [comment]
Close this PR and create a new one using @ganeshmurthy deprecationName

**Pull request closed without merging**